frontend/src/models/protocols.ts

```diff
@@ -0,0 +1,26 @@
+
+/* Defines enum values for the protocol names. */
+export enum ProtocolNames {
```

Comment:
don't copy the DB in the frontend ... I doubt you need all those names mapped in the frontend. Keep to what is used. We don't want to keep updating those if they don't have a use.

Reply:
The protocol name is the only way for the frontend to know which protocol is which, so I created some constants for the names, in case the names change in the future and to avoid silly typo bugs.

Reply:
This is not ideal but yeah i see ...